### PR TITLE
Adds in the Mining scanner

### PR DESCRIPTION
**Pull request by MaxIsJoe**
The mining scanner is a tool used to highlight ores for miners during their job. 
It takes 20 seconds per each use (5 for advaned) and all highlights are done client side.

https://user-images.githubusercontent.com/34368774/179123403-a9e7c084-61b6-428c-9754-867ce211319f.mp4

### Notes:

- This isn't automatic like on /tg/ and requires the player to activate it every now and then to scan ores.
- This will only highlight stuff that are 25 tiles near the player for performance reasons
- There is sadly no way to show anything outside of the FOV without using lights
- - Though im proud of how it turned out to be and how it's animated :D
- Uses placeholder assets for its sprites until we port the scanner sprites from /tg/
- I left in the ability to upgrade the thing later when we have the proper prefabs or the ability for science to upgrade items
- Needs more testing

# Changelog:

CL: [New] - Adds in the mining scanner.

### CODE REVIEW

**Comment by AtnerNT** [comment]
Im pretty sure w use a custom material to be blocked by the FOV, you could try change the material of highlighted ores so it is exluded from the FoV. Especially this only client side

**Comment by MaxIsJoe** [comment]
I tried that, failed.

If you know how it can be setup properly then PR it to my repo before this gets added

**Comment by AtnerNT** [comment]
Damn, I took a look and materials are per tile map not per tile which is a shame. However you misspelt mining in the scanner name, might jsut wanna fi that before you merge the PR

**Comment by MaxIsJoe** [comment]
Done. Merge pls

File: UnityProject/Assets/Scripts/Items/Cargo/MiningScanner.cs
Comment (Alainx277) [inline UnityProject/Assets/Scripts/Items/Cargo/MiningScanner.cs:21]: Can this be removed?

File: UnityProject/Assets/Scripts/Items/Cargo/MiningScanner.cs
Comment (Alainx277) [inline UnityProject/Assets/Scripts/Items/Cargo/MiningScanner.cs:40]: Does just calling the RPC not work in the editor?

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/Layer.cs
Comment (Alainx277) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/Layer.cs]: Can you put a short comment above? It's a bit too dense to see what it does at a glance.

File: UnityProject/Assets/Scripts/Tilemaps/Tiles/SimpleTile.cs
Comment (Alainx277) [inline UnityProject/Assets/Scripts/Tilemaps/Tiles/SimpleTile.cs:21]: I think a tooltip would be nice.

**Review by Alainx277 — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/Items/Cargo/MiningScanner.cs
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Cargo/MiningScanner.cs:21]: I'd like to leave this here because I still want to do some tests when this gets merged on staging

File: UnityProject/Assets/Scripts/Items/Cargo/MiningScanner.cs
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Cargo/MiningScanner.cs:40]: it doesn't

File: UnityProject/Assets/Scripts/Tilemaps/Tiles/SimpleTile.cs
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Tilemaps/Tiles/SimpleTile.cs:21]: You can't have both NaughtyAttributes and a unity attribute at the same time, it breaks it

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/Layer.cs
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/Layer.cs]: Done

File: UnityProject/Assets/Scripts/Tilemaps/Tiles/SimpleTile.cs
Comment (Alainx277) [reply UnityProject/Assets/Scripts/Tilemaps/Tiles/SimpleTile.cs:21]: Sad.

**Review by Alainx277 — APPROVED** [review]
Code-wise ready to be merged.

**Pull request merged**
